ocs/ocs_feed.py

```diff
@@ -136,6 +138,7 @@ def flush_buffer(self):
                     {k: b.encoded() for k, b in self.blocks.items() if b.timestamps},
                     self.encoded()
                 ))
+                self.message_idx += 1
```

Comment:
This should probably not be inside the try.  The blocks are going to get cleared, so you'll want to bump message_idx.

Reply:
My intention was for the `message_idx` to increment for each successfully published message, rather than for each attempted message, but am open to changing this if there is a good reason.

I believe there's not a huge difference between the two modes... The only difference I can think of now is that if the receiver observes an index skip, they can be confident that its not an issue on the sender's side, and that it's a network / crossbar problem.

Reply:
I guess the counterpoint is that if we move it out of the try then the receiver can know whether they're missing data that the agent intended to send, possibly due to agent-side issues.

Reply:
> My intention was for the `message_idx` to increment for each successfully published message, rather than for each attempted message, but am open to changing this if there is a good reason.
> 
> I believe there's not a huge difference between the two modes... The only difference I can think of now is that if the receiver observes an index skip, they can be confident that its not an issue on the sender's side, and that it's a network / crossbar problem.

But is a failure at this point in the agent a guarantee that the message won't make it to a consumer?  I know that's _likely_ the case, but I'm doubtful that it's guaranteed.  It is much safer to never risk sending the same message_idx out on the wire.

Reply:
ok ya I see, so the increment should probably happen before the try statement, or in a finally block

Reply:
or like you say, after the encoding. Having the increment happen in the encoding function also could be right, but it feels weird to me since I don't love having the encode function change the feed state, even if its behind an optional kwarg.